### ISSUE TITLE
[tune] String summarization/representations for user objects

### ISSUE THREAD

**Issue by richardliaw** [ray-project/ray#10489]
### Describe your feature request 

This includes, but is not limited to:

1. Trial object (`trial.summary()`)
2. Experiment object
3. TrialCheckpoint object
4. Analysis object

Otherwise, it's quite hard to parse + operate over results.

**Comment by cszhu** [comment]
Hello! This P2 issue has seen no activity in at least 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
